examples.py/3D/Form/bricktower.py

```diff
@@ -15,35 +15,36 @@
 def setup():
     size(640, 360, OPENGL)
 
+
 def draw():
-  background(0)
-  (tempX, tempY, tempZ) = (0, 0, 0)
-  fill(182, 62, 29)
-  noStroke()
-  # Add basic light setup
-  lights()
-  translate(width / 2, height * 1.2, -380)
-  # Tip tower to see inside
-  rotateX(radians(-45))
-  # Slowly rotate tower
-  rotateY(frameCount * PI / 600)
-  for i in range(brickLayers):
+    background(0)
+    (tempX, tempY, tempZ) = (0, 0, 0)
+    fill(182, 62, 29)
+    noStroke()
+    # Add basic light setup
+    lights()
+    translate(width / 2, height * 1.2, -380)
+    # Tip tower to see inside
+    rotateX(radians(-45))
+    # Slowly rotate tower
+    rotateY(frameCount * PI / 600)
+    for i in xrange(brickLayers):
     # Increment rows
-    tempY -= brickHeight
-    # Alternate brick seams
-    angle = 360.0 / bricksPerLayer * i / 2
-    for j in range(bricksPerLayer):
-      tempZ = cos(radians(angle)) * radius
-      tempX = sin(radians(angle)) * radius
-      pushMatrix()
-      translate(tempX, tempY, tempZ)
-      rotateY(radians(angle))
-      # Add crenelation
-      if (i == brickLayers - 1):
-        if (j % 2 == 0):
-          box(brickWidth, brickHeight, brickDepth)
-      else:
-        # Create main tower
-        box(brickWidth, brickHeight, brickDepth)
-      popMatrix()
-      angle += 360.0 / bricksPerLayer
+        tempY -= brickHeight
+        # Alternate brick seams
+        angle = 360.0 / bricksPerLayer * i / 2
+        for j in xrange(bricksPerLayer):
+            tempZ = cos(radians(angle)) * radius
+            tempX = sin(radians(angle)) * radius
+            pushMatrix()
+            translate(tempX, tempY, tempZ)
+            rotateY(radians(angle))
+            # Add crenelation
+            if (i == brickLayers - 1):
+                if (j % 2 == 0):
+                    box(brickWidth, brickHeight, brickDepth)
+            else:
+                # Create main tower
+                box(brickWidth, brickHeight, brickDepth)
```

Comment:
The only time `box(brickWidth, brickHeight, brickDepth)` doesn't happen is when:
`i == brickLayers - 1 and j % 2 == 1`

IMO, it's harder to figure that logic out by looking at the big nested if/if/else block.
I had to study it for some minutes in order to come up w/ the actual synthesized conditional expression!  O_o